### PR TITLE
fix: reject operation multisig

### LINKED ISSUE #1774: Multisig operations. Can't reject operation if has a proxy
## Description

User can't reject the operation, confirmation page is empty, if user has proxy for signatory account.

<details>
  <summary>Example</summary>

https://github.com/novasamatech/nova-spektr/assets/40560660/9942438f-5754-4582-a377-0849bd5be39a

</details>

## Steps to Reproduce

1. Add 2 wallets in spektr, signatory_a and signatory_b
2. Create multisig wallet with 2 signatory, signatory_a and signatory_b
3. Create multisig transaction from signatory_a
4. Create proxy from signatory_ on signatory_b
5. Try to reject multisig operation

## Expected Behavior
Should be taken original signatory_a account, successfull reject

## Actual Behavior
May be taken a proxy account on confirmation screen, that lead to empty confirmation screen

Screenshot 2024-05-20 at 12 57 19

## Environment

- Operating system: macos
- Version: 1.7.3

### LINKED ISSUE #2595: Multisig. Can't reject operation via proxy
## Description

When user has a proxy for multisig's signatory, reject operation can't be signed.

https://github.com/user-attachments/assets/e0814bf9-f8db-429b-861c-5bd2f43ad594

Account setup:
- [accounts.json](https://github.com/user-attachments/files/17674289/accounts.json)
- [wallets.json](https://github.com/user-attachments/files/17674290/wallets.json)

## Steps to Reproduce

1. Add account_A
2. Create proxy from account_B to account_A
3. Create multisig with account_A\account_B\account_C
4. Create an operation from account_B via account_A using proxy
5. Go to multisig operations
6. Try to reject the operation

## Expected Behavior
Successfully reject the operation via proxy

## Actual Behavior
Can't reject an operation - empty signing page

## Environment
https://github.com/novasamatech/nova-spektr/commit/9761c24c9fc3b3e4612dc41a53c44680cff488c5

### LINKED ISSUE #2611: Multisig. Operations. Reject confirm screen - broken indent
## Description

Screenshot 2024-11-11 at 13 47 24

## Environment
https://github.com/novasamatech/nova-spektr/commit/1b29836ffc7ca408a44df236bae6f0d8b98bf087

### PR DESCRIPTION
**Pull request by sokolova-an**
Closes https://github.com/novasamatech/nova-spektr/issues/2595 \ https://github.com/novasamatech/nova-spektr/issues/1774

### CODE REVIEW

File: src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx [hunk 1]
```diff
@@ -17,29 +17,32 @@ import {
...
-export const Confirmation = ({ tx, account, connection, signatory, feeTx, onSign }: Props) => {
+export const Confirmation = ({ tx, account, chainConnection, signAccount, feeTx, onSign }: Props) => {
   const { t } = useI18n();
   const [isFeeLoaded, setIsFeeLoaded] = useState(false);
   const fiatFlag = useUnit(priceProviderModel.$fiatFlag);
 
   const wallets = useUnit(walletModel.$wallets);
-  const wallet = wallets.find((w) => w.id === signatory?.walletId);
+  const signerWallet = walletUtils.getWalletFilteredAccounts(wallets, {
+    walletFn: walletUtils.isValidSignSignatory,
```
Comment (Asmadek) [inline src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx:40]: should be `isValidSignSignatory`

**Review by Asmadek — APPROVED** [review]

File: src/renderer/entities/transaction/lib/transactionBuilder.ts [hunk 2]
```diff
@@ -478,3 +480,27 @@ function buildRemoveVotes({ chain, accountId, votes }: RemoveVotesParams): Trans
...
+  tx: MultisigTransactionDS;
+};
+function buildRejectMultisigTx({ chain, signerAddress, threshold, otherSignatories, tx }: RejectTxParams): Transaction {
+  return {
+    chainId: chain.chainId,
+    address: signerAddress,
+    type: TransactionType.MULTISIG_CANCEL_AS_MULTI,
+    args: {
+      threshold: threshold,
+      otherSignatories: otherSignatories.sort(),
```
Comment (johnthecat) [inline src/renderer/entities/transaction/lib/transactionBuilder.ts]:
```suggestion
      otherSignatories: Array.from(otherSignatories).sort(),
```

File: src/renderer/pages/Operations/components/modals/ApproveTx.tsx [hunk 3]
```diff
@@ -237,29 +235,36 @@ const ApproveTx = ({ tx, account, connection }: Props) => {
...
-        isOpen={activeStep !== Step.SUBMIT && isModalOpen}
-        title={<OperationTitle title={t(transactionTitle, { asset: asset?.symbol })} chainId={tx.chainId} />}
-        contentClass={activeStep === Step.SIGNING ? '' : undefined}
-        headerClass="py-3 pl-5 pr-3"
-        panelClass="w-[440px]"
-        onClose={handleClose}
-      >
+    <Modal size="md" onToggle={handleClose}>
+      <Modal.Trigger>
+        <Button className="ml-auto">{t('operation.approveButton')}</Button>
```
Comment (johnthecat) [inline src/renderer/pages/Operations/components/modals/ApproveTx.tsx]: You can pass button as children inside ApproveTxModal component, it should be clearer for reading

File: src/renderer/pages/Operations/components/modals/RejectTx.tsx [hunk 4]
```diff
@@ -180,30 +165,41 @@ const RejectTx = ({ tx, account, connection }: Props) => {
...
         onClose={handleClose}
-      >
+      />
+    );
+  }
+
+  return (
+    <Modal size="md" onToggle={handleClose}>
+      <Modal.Trigger>
+        <div className="flex justify-between">
```
Comment (johnthecat) [inline src/renderer/pages/Operations/components/modals/RejectTx.tsx]: Same here

File: src/renderer/entities/transaction/lib/transactionBuilder.ts (hunk 2 above)
Comment (sokolova-an) [reply src/renderer/entities/transaction/lib/transactionBuilder.ts]: why add this? it's already an array

Comment (johnthecat) [reply src/renderer/entities/transaction/lib/transactionBuilder.ts]: `sort` method creates array mutation, which can lead to unexpected behavior

Comment (sokolova-an) [reply src/renderer/entities/transaction/lib/transactionBuilder.ts]: Hm, in this case, it's better not to add it here in the builder at all, instead, move it to where it’s passed to the function (and be sure that we not mutating anything important with that) 
I'm not sure if we even need to sort it

**Review by johnthecat — APPROVED** [review]

**Pull request merged**